src/physics_common.cpp

```diff
@@ -23,4 +24,155 @@ void russian_roulette(Particle* p)
   }
 }
 
+//==============================================================================
+// IMPORTANCE_SPLIT
+//==============================================================================
+
+void split_by_ratio(Particle* p, double importance_ratio) {
+  p->wgt_last_ = p->wgt_;
+  double importance_rounded = std::floor(importance_ratio);
+  
+  int num_splits = 0;
+  // how many splits - probabilistic 3/2 split 
+  //  half the time we get 3 and half we get 2
+  if ( prn() < importance_ratio - importance_rounded )
+    num_splits = int(importance_rounded);
+  else 
+    num_splits = int(std::ceil(importance_ratio));
```

Comment:
We generally use braces for non-trivial `if` blocks:
```C++
if (prn() < importance_ratio - importance_rounded) {
  num_splits = int(importance_rounded);
} else {
  num_splits = int(std::ceil(importance_ratio));
}
```

Reply:
Also, shouldn't the logic be reversed? If important_ratio = 2.9, this would give num_splits=2 90% of the time rather than num_splits=3 90% of the time.